### DISCUSSION TITLE
How to edit object values from a sibling controller?

### DISCUSSION THREAD

**Discussion by alexw23** [react-hook-form/react-hook-form#5303] [Q&A] [answered]
I notice that when the state changes it doesn't set the existing values on the new inputs, if the new inputs are part of the nested object.

Sandbox: https://codesandbox.io/s/react-hook-form-get-started-forked-rd9t0?file=/src/index.js

Desired result:
1) Select ice cream Chocolate
2) Toggle manual entry
3) See that first input contains "chocolate"
4) See that second input contains "good"

**Comment by bluebill1049** [comment, marked as answer]
I would do something like the below: 

https://codesandbox.io/s/react-hook-form-get-started-forked-7jfzf?file=/src/index.js

I would convert what I got into a controlled component and host these two inputs. I am just been lazy and quick in above.

**Reply by alexw23** [reply]
I like it - very clean approach, how would you handle validation and focusing on errors though?

**Reply by alexw23** [reply]
I ended up making a modal, and creating a new form. Then upon modal close triggering onChange within the main controller. Just seemed like a cleaner flow overall, and clearer separation of concerns.

**Reply by bluebill1049** [reply]
That sounds great and better UX as well.
